### PR TITLE
Add a page describing what is GDNative

### LINKED ISSUE #1024: An "About GDNative" document
So we now have the GDNative C tutorial, which is an excellent addition to the docs, however...the docs don't really explain what GDNative itself *is* (nevermind the fact that said tutorial is kinda tucked away in Community::Tutorials::GDNative right now, so it's barely visible - see #1020), and I think it would be something worth doing since it's not a very 'intuitive' (at first) feature, yet it's quite incredible with how much potential it has.

I do think that a lot of info could be lifted wholecloth from the [GDNative blog post](https://godotengine.org/article/look-gdnative-architecture) for such an article, as well as for documenting some of the GDNative-related classes in the API reference (which is a bit beyond the scope of this issue, but is nonetheless related). 
I don't know how much I'd be able to help out with this myself, since GDNative is ...a bit lower-level than I'm comfortable with documenting.
 
@karroffel @BastiaanOlij

### PR DESCRIPTION
**Pull request by Calinou**
This closes #1024.

### CODE REVIEW

File: tutorials/plugins/gdnative/what_is_gdnative.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+Third-party
+^^^^^^^^^^^
+
+The bindings below are developed by the community and are sorted by alphabetical order.
+
+- `D <https://github.com/godot-d/godot-d>`__
+- `Kotlin <https://github.com/utopia-rise/godot-kotlin>`__
+- `Nim <https://github.com/pragmagic/godot-nim>`__
+- `Rust <https://github.com/godot-rust/godot-rust>`__
```
Comment (StraToN) [inline tutorials/plugins/gdnative/what_is_gdnative.rst:81]: Python could be added to this list I think: https://github.com/touilleMan/godot-python

**Review by StraToN — DISMISSED** [review]

**Comment by akien-mga** [comment]
CC @karroffel @bojidar-bg @vnen

**Comment by NathanLovato** [comment]
The article is clear. Thanks for your work! With some edits done, I'll merge, and if someone feels it's missing something, comments, issues, and PRs are welcome.

**Comment by Calinou** [comment]
For future reference, this page was cherry-picked to the `3.3` branch by https://github.com/godotengine/godot-docs/commit/b4476bd04e9f658373a0af485589fb7dfc2e6c33.

**Pull request merged**